www/include/views/graphs/exportData/ExportCSVServiceData.php

```diff
@@ -98,17 +126,27 @@ function get_error($str)
 $datas = array();
 $listEmptyMetric = array();
 
-$query = "SELECT DISTINCT metric_id, metric_name FROM metrics, index_data " .
-    "WHERE metrics.index_id = index_data.id AND id = '$index' ORDER BY metric_name";
-$DBRESULT = $pearDBO->query($query);
-while ($index_data = $DBRESULT->fetchRow()) {
+$stmt = $pearDBO->prepare(
+    'SELECT DISTINCT metric_id, metric_name ' .
+    'FROM metrics, index_data ' .
+    'WHERE metrics.index_id = index_data.id AND id = :index ORDER BY metric_name'
+);
+
+$stmt->bindValue(':index', $index, \PDO::PARAM_INT);
+$stmt->execute();
+
+while ($index_data = $stmt->fetchRow()) {
     $listMetric[$index_data["metric_id"]] = $index_data["metric_name"];
     $listEmptyMetric[$index_data["metric_id"]] = '';
-    $query2 = "SELECT ctime,value FROM data_bin WHERE id_metric = '" . $index_data["metric_id"] .
-        "' AND ctime >= '" . htmlentities($_GET["start"], ENT_QUOTES, "UTF-8") .
-        "' AND ctime < '" . htmlentities($_GET["end"], ENT_QUOTES, "UTF-8") . "'";
-    $DBRESULT2 = $pearDBO->query($query2);
-    while ($data = $DBRESULT2->fetchRow()) {
+    $stmt2 = $pearDBO->prepare(
+        "SELECT ctime,value FROM data_bin WHERE id_metric = ':metricId' " .
+        "AND ctime >= ':start' AND ctime < ':end'"
+    );
+    $stmt2->bindValue(':start', $start, \PDO::PARAM_INT);
```

Comment:
what will happen is $start or $end is false or null ?

Reply:
I guess I can check before if does are defined